proposals/NNNN-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,239 @@
+struct BoilingPoint<T> {
+  init(wrappedValue: T) {
+    self.wrappedValue = wrappedValue
+  }
+
+  var wrappedValue: T
+  var projectedValue: T
+  
+  func toFarenheit() {}
+}
+
+struct Temperature {
+  @BoilingPoint var water: Double = 373.1
+}
+```
+
+The initial value for `water` is set using the `init(wrappedValue:)` initializer of the `BoilingPoint` property wrapper type. To ensure that `water` is not changed again, we could add logic to the `wrappedValue` setter to check if the property was already initialized and prevent re-assignment. However, this is an inconvenient solution.
+
+Instead, we could declare the `water` property with a `let`, synthesize a local `let` constant for the backing storage property (prefixed with an underscore), and only allow the property wrapper to be initialized once, passing the assigned value to `init(wrappedValue:)`. Now, rewriting `Temperature`’s properties as `let` constants will translate to:
+```swift
+struct Temperature {
+  @BoilingPoint let water: Double = 373.1
+
+  // ... introduces _water stored property
+  private let _water: BoilingPoint<Double> = BoilingPoint<Double>(wrappedValue: 373.1)
+  
+  // ... getter-only computed property
+  @BoilingPoint var water: Double {
+    get { return self._water.wrappedValue }
+  }
+  
+  // ... getter-only projectedValue property
+  internal var $water: Double {
+    get { return self._water.projectedValue }
+  }
+}
+```
+
+Declaring a `let` wrapped property of `water` of type `BoilingPoint` generates a `let` declared storage property of `_water`, a getter-only computed property that assigns the declared value to the backing property wrapper's `wrappedValue`. A let declared wrapped property also generates `$water`, a getter-only projectedValue property.
+
+The `let` declared property can also be initialized once via the generated storage property. For example, reusing the above example:
+```swift
+  @BoilingPoint let water: Double
+  
+  init() {
+    _water = BoilingPoint(wrappedValue: 373.1)
+  }
+```
+
+After the `let` declared property has been initialized, attempting to reassign it will show an error. Clients can continue to manipulate the projection just like with `var` declared properties.
+```swift
+  temperature.water = 400.0 // error: cannot assign to property: 'water' is a 'let' constant
+  temperature.$water.toFarenheit() // converts to 212.0
+```
+
+Property wrappers with `let` declarations will be allowed both as members and local declarations, as envisioned by [SE-0258](https://github.com/apple/swift-evolution/blob/main/proposals/0258-property-wrappers.md) for `var` declared property wrappers. All other property wrapper traits also remain unchanged from SE-0258.
+
+## Detailed Design
+
+Marking a property wrapped declaration with a `let` makes that property a getter-only computed property and introduces a `let` synthesized stored property whose type is the wrapper type. This expands where and how property wrappers can be used in Swift.
```

Comment:
```suggestion
```

This sentence repeats the statements above. It doesn't seem to add any new information.